crates/dekaf/src/topology.rs

```diff
@@ -94,8 +93,45 @@ impl Collection {
         let validator = doc::Validator::new(json_schema)?;
         let mut shape = doc::Shape::infer(&validator.schemas()[0], validator.schema_index());
 
+        // In order for the `_meta/is_deleted` field to get serialized correctly into avro, we need an explicit
+        // field definition for it in the schema. Since some schemas can have `additionalProperties: true`
+        // (or unset, which defaults to true), trying to use e.g Shape::widen() or Shape::union() will
```

Comment:
nit: i know this comment is relevant to why you were just in here, but it's not super relevant to the reader. Why would they presume `Shape::widen()` is preferred in the first place?

Reply:
Heh, I keep thinking this code is smelly so I keep trying to justify its existence vs a much simpler call to e.g `widen()` or `union()`. I'll get rid of the comment